secure-api-gateway-ob-uk-rcs-consent-store/secure-api-gateway-ob-uk-rcs-consent-store-client/src/main/java/com/forgerock/sapi/gateway/rcs/consent/store/client/payment/vrp/BaseRestDomesticVRPConsentStoreClient.java

```diff
@@ -0,0 +1,79 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.rcs.consent.store.client.payment.vrp;
+
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.forgerock.sapi.gateway.rcs.consent.store.client.BaseRestConsentStoreClient;
+import com.forgerock.sapi.gateway.rcs.consent.store.client.ConsentStoreClientConfiguration;
+import com.forgerock.sapi.gateway.rcs.consent.store.client.ConsentStoreClientException;
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.RejectConsentRequest;
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.payment.AuthorisePaymentConsentRequest;
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.payment.vrp.v3_1_10.CreateDomesticVRPConsentRequest;
+import com.forgerock.sapi.gateway.rcs.consent.store.datamodel.payment.vrp.v3_1_10.DomesticVRPConsent;
+import com.forgerock.sapi.gateway.uk.common.shared.api.meta.obie.OBVersion;
+
+/**
+ * Implementation of the DomesticVRPConsentStoreClient which makes REST calls over HTTP
+ */
+public class BaseRestDomesticVRPConsentStoreClient extends BaseRestConsentStoreClient implements DomesticVRPConsentStoreClient {
+
+    private final String consentServiceBaseUrl;
+
+    public BaseRestDomesticVRPConsentStoreClient(ConsentStoreClientConfiguration consentStoreClientConfiguration, RestTemplateBuilder restTemplateBuilder,
+                                                 ObjectMapper objectMapper, OBVersion obVersion) {
+        super(restTemplateBuilder, objectMapper);
+        this.consentServiceBaseUrl = consentStoreClientConfiguration.getBaseUri() + "/v" + obVersion.getCanonicalVersion() + "/domestic-vrp-consents";
+    }
+
+    @Override
+    public DomesticVRPConsent createConsent(CreateDomesticVRPConsentRequest createConsentRequest) throws ConsentStoreClientException {
+        final HttpEntity<CreateDomesticVRPConsentRequest> requestEntity = new HttpEntity<>(createConsentRequest, createHeaders(createConsentRequest.getApiClientId()));
+        return doRestCall(consentServiceBaseUrl, HttpMethod.POST, requestEntity, DomesticVRPConsent.class);
+    }
+
+    @Override
+    public DomesticVRPConsent getConsent(String consentId, String apiClientId) throws ConsentStoreClientException {
+        final String url = consentServiceBaseUrl + "/" + consentId;
```

Comment:
Sorry to keep asking...

Can this `consentId` be null here - as you'd end up with an unusual URL.

Reply:
Siilar elsewhere... I'm assuming this isn't possible.

Reply:
Yes, the consentId can be null but in general this is not an issue because we have handling around this - if the consent is not found, we will thrown a consent not found exception if it's null or has an invalid value provided.